core/keyboard_navigation_controller.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * The KeyboardNavigationController handles coordinating Blockly-wide
+ * keyboard navigation behavior, such as enabling/disabling full
+ * cursor visualization.
+ */
+export class KeyboardNavigationController {
+  /** Whether the user is actively using keyboard navigation. */
+  private isActive = false;
+  /** Css class name added to body if keyboard nav is active. */
+  private activeClassName = 'blocklyKeyboardNavigation';
+
+  /**
+   * Sets whether we think the user is actively using keyboard navigation.
+   *
+   * If they are, we'll apply a css class to the entire page so that
+   * focused items can apply additional styling for keyboard users.
+   *
+   * @param isUsing
```

Comment:
Did you mean to document this?

Reply:
We don't require documentation for trivial parameters. You have to list the `@param` so that it shows up properly in the generated reference docs, but adding "whether the user is using keyboard navigation" doesn't really add anything of value that isn't obvious from the name and the other documentation, so it's not required. I can add it if you think it is useful though.

Reply:
Nope, that explanation makes sense to me. I haven't spent a lot of time actually understanding how the generated docs look in different scenarios, so this is helpful context for me.